src/main/webapp/app/services/visualizations/vizualization-render.service.js

```diff
@@ -126,17 +126,23 @@
                 visualMetadata,
                 el,
                 panel,
-                isNotification
+                isNotification,
+                isIframe
             );
         }
 
-        function setMetaData(v, metadata, contentId, isNotification) {
+        function setMetaData(v, metadata, contentId, isNotification,isIframe) {
             if (v) {
                 addWidgets();
                 vm.data = v;
                 vm.data.data = metadata.data;
                 vm.widget = v.metadataVisual.functionname;
-                createWidget(vm.data, contentId, isNotification);
+                if(vm.widget==="GenerateIframe"){
```

Comment:
Is this condition required when u have isIframe var contains true/false? You can just pass isIframe only